### ISSUE TITLE
Concurrent docker swarm service name collisions

### ISSUE THREAD

**Issue by kislyuk** [chanzuckerberg/miniwdl#368]
Sorry if we covered this before. I’m getting what looks like a docker swarm concurrent namespace use collision:
```
2020-04-09 21:46:56+00:00 2020-04-09 21:46:56.782 wdl.w:idseq_host_filter.download1.t:download-s3parcp ERROR task s3parcp ((buffer) Ln 2 Col 1) failed :: dir: "/mnt/20200409_214655_834_idseq_host_filter/download/1", error: "APIError", message: "409 Client Error: Conflict (\"rpc error: code = AlreadyExists desc = name conflicts with an existing object: service wdl-29-1-download-s3parcp already exists\")"
```
Is there a way to configure miniwdl to name the docker tasks with a prefix unique to the container?

**Comment by kislyuk** [comment]
It appears https://github.com/chanzuckerberg/miniwdl/blob/master/WDL/runtime/task.py#L447 was already supposed to be hardened against this, but it does not seem to be working.

**Comment by kislyuk** [comment]
Can we add hostname as a dash-separated component at the beginning of this name? Hostname in a docker container is that container's ID.

**Comment by mlin** [comment]
Yeah, the new kink is when we have miniwdl running in two different docker containers, the processes are liable to see themselves having the same PID due to the separate namespaces. Hostname might be too long in the more common case when miniwdl itself isn't running inside docker. I'll poke around for something to use

**Comment by mlin** [comment]
@kislyuk as a cheap workaround, is it possible to set this pidMode flag in ECS? https://aws.amazon.com/about-aws/whats-new/2018/10/amazon-ecs-now-allows-two-additional-docker-flags-/

**Comment by mlin** [comment]
terraform: https://www.terraform.io/docs/providers/aws/r/ecs_task_definition.html#pid_mode

**Comment by kislyuk** [comment]
Unfortunately, Batch does not provide access to this parameter. You can see the list of all task definition parameters Batch provides access to here: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/batch.html#Batch.Client.register_job_definition

**Comment by mlin** [comment]
https://github.com/chanzuckerberg/miniwdl/commit/dd154b67acdf4431cc4aaa33a492e79b791b0ff3 should cudgel this by just slapping 120 bits of entropy onto the ID. (Appending it to the right-hand side helps the `docker service ls` display remain useful)

**Comment by mlin** [comment]
released v0.7.3
